internal/indexer/indexer.go

```diff
@@ -127,8 +148,54 @@ func (i *Indexer) Index() error {
 	return nil
 }
 
+func (i *Indexer) startImportMonikerReferenceTracker(wg *sync.WaitGroup) {
+	go func() {
+		contained := struct{}{}
+
+		wg.Add(1)
+		for {
+			nextReference := <-i.importMonikerChannel
```

Comment:
You should be able to do `for nextReference := range i.ilmportMonikerChannel`, which reads until the channel is closed. This will get rid of the `done` field in your reference struct and simplify the condition below as well.